### PR TITLE
K8SPG-372: Align cluster-wide support with other operators

### PR DESCRIPTION
**Pull request by inelpandzic**
[K8SPG-372](https://jira.percona.com/browse/K8SPG-372) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

**CHANGE DESCRIPTION**
---
**Problem:**
We inherited Crunchy env vars to handle cluster-wide mode (PGO_TARGET_INSTANCE) and the only support was to watch all the namespaces or just the one in which the operator runs.

**Solution:**
We want to align CW with our other operators, that is to have the same behavior and the same way of configuring it, that is with `WATCH_NAMESPACE` env var.

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?
- [x] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [x] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [x] Is an E2E test/test case added for the new feature/change?
- [x] Are unit tests added where appropriate?

**Config/Logging/Testability**
- [x] Are all needed new/changed options added to default YAML files?
- [x] Are the manifests (crd/bundle) regenerated if needed?
- [x] Did we add proper logging messages for operator actions?
- [x] Did we ensure compatibility with the previous version or cluster upgrade process?
- [x] Does the change support oldest and newest supported PG version?
- [x] Does the change support oldest and newest supported Kubernetes version?

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| demand-backup | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/1c7d67d41/demand-backup.log) |
| init-deploy | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/1c7d67d41/init-deploy.log) |
| monitoring | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/1c7d67d41/monitoring.log) |
| operator-self-healing | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/1c7d67d41/operator-self-healing.log) |
| scaling | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/1c7d67d41/scaling.log) |
| scheduled-backup | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/1c7d67d41/scheduled-backup.log) |
| self-healing | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/1c7d67d41/self-healing.log) |
| start-from-backup | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/1c7d67d41/start-from-backup.log) |
| telemetry-transfer | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/1c7d67d41/telemetry-transfer.log) |
| users | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/1c7d67d41/users.log) |
| We run 10 out of 10|

commit: https://github.com/percona/percona-postgresql-operator/pull/529/commits/1c7d67d41daf5b45ca3de64ce172ad4417d29f12
image: `perconalab/percona-postgresql-operator:PR-529-1c7d67d41`

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| demand-backup | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/1c7d67d41/demand-backup.log) |
| init-deploy | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/1c7d67d41/init-deploy.log) |
| monitoring | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/1c7d67d41/monitoring.log) |
| operator-self-healing | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/1c7d67d41/operator-self-healing.log) |
| scaling | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/1c7d67d41/scaling.log) |
| scheduled-backup | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/1c7d67d41/scheduled-backup.log) |
| self-healing | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/1c7d67d41/self-healing.log) |
| start-from-backup | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/1c7d67d41/start-from-backup.log) |
| telemetry-transfer | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/1c7d67d41/telemetry-transfer.log) |
| users | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/1c7d67d41/users.log) |
| We run 10 out of 10|

commit: https://github.com/percona/percona-postgresql-operator/pull/529/commits/1c7d67d41daf5b45ca3de64ce172ad4417d29f12
image: `perconalab/percona-postgresql-operator:PR-529-1c7d67d41`

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/874fbe7b8/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/874fbe7b8/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/874fbe7b8/monitoring.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/874fbe7b8/operator-self-healing.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/874fbe7b8/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/874fbe7b8/scheduled-backup.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/874fbe7b8/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/874fbe7b8/start-from-backup.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/874fbe7b8/telemetry-transfer.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/874fbe7b8/users.log) |
| We run 10 out of 10|

commit: https://github.com/percona/percona-postgresql-operator/pull/529/commits/874fbe7b85455e6312044048732680acfd251db2
image: `perconalab/percona-postgresql-operator:PR-529-874fbe7b8`

File: deploy/operator.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ spec:
     spec:
       containers:
       - env:
-        - name: PGO_TARGET_NAMESPACE
+        - name: WATCH_NAMESPACE
```
[the remark this answers is not in the document]
Comment (inelpandzic) [reply deploy/operator.yaml:33]: Good question, but I don't think we need PGO_TARGET_NAMESPACE env var in our operator and for Crunchy's as well. This env var is used only in one place when configuring the manger. And now we do it in our way with our WATCH_NAMESPACE env var.

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/b759e295b/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/b759e295b/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/b759e295b/monitoring.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/b759e295b/operator-self-healing.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/b759e295b/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/b759e295b/scheduled-backup.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/b759e295b/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/b759e295b/start-from-backup.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/b759e295b/telemetry-transfer.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/b759e295b/users.log) |
| We run 10 out of 10|

commit: https://github.com/percona/percona-postgresql-operator/pull/529/commits/b759e295bc317da0f1f5d2d845166a90e82154aa
image: `perconalab/percona-postgresql-operator:PR-529-b759e295b`

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/fe98365c3/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/fe98365c3/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/fe98365c3/monitoring.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/fe98365c3/operator-self-healing.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/fe98365c3/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/fe98365c3/scheduled-backup.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/fe98365c3/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/fe98365c3/start-from-backup.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/fe98365c3/telemetry-transfer.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/fe98365c3/users.log) |
| We run 10 out of 10|

commit: https://github.com/percona/percona-postgresql-operator/pull/529/commits/fe98365c34209d712afa67ab5241791779cb7c4e
image: `perconalab/percona-postgresql-operator:PR-529-fe98365c3`

**Review by egegunes — APPROVED** [review]

**Review by hors — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/6fd1c68c7/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/6fd1c68c7/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/6fd1c68c7/monitoring.log) |
| operator-self-healing | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/6fd1c68c7/operator-self-healing.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/6fd1c68c7/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/6fd1c68c7/scheduled-backup.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/6fd1c68c7/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/6fd1c68c7/start-from-backup.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/6fd1c68c7/telemetry-transfer.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/6fd1c68c7/users.log) |
| We run 10 out of 10|

commit: https://github.com/percona/percona-postgresql-operator/pull/529/commits/6fd1c68c71ac5c0f2a30a0e12c049121bdd5ac47
image: `perconalab/percona-postgresql-operator:PR-529-6fd1c68c7`

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| demand-backup | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/24f16bc19/demand-backup.log) |
| init-deploy | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/24f16bc19/init-deploy.log) |
| monitoring | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/24f16bc19/monitoring.log) |
| operator-self-healing | [failure](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/24f16bc19/operator-self-healing.log) |
| scaling | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/24f16bc19/scaling.log) |
| scheduled-backup | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/24f16bc19/scheduled-backup.log) |
| self-healing | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/24f16bc19/self-healing.log) |
| start-from-backup | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/24f16bc19/start-from-backup.log) |
| telemetry-transfer | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/24f16bc19/telemetry-transfer.log) |
| users | [skipped](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/24f16bc19/users.log) |
| We run 4 out of 10|

commit: https://github.com/percona/percona-postgresql-operator/pull/529/commits/24f16bc19127960fcfa4634b2eb261fc4a50eec6
image: `perconalab/percona-postgresql-operator:PR-529-24f16bc19`

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/24f16bc19/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/24f16bc19/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/24f16bc19/monitoring.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/24f16bc19/operator-self-healing.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/24f16bc19/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/24f16bc19/scheduled-backup.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/24f16bc19/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/24f16bc19/start-from-backup.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/24f16bc19/telemetry-transfer.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-529/24f16bc19/users.log) |
| We run 10 out of 10|

commit: https://github.com/percona/percona-postgresql-operator/pull/529/commits/24f16bc19127960fcfa4634b2eb261fc4a50eec6
image: `perconalab/percona-postgresql-operator:PR-529-24f16bc19`

**Pull request merged**